packages/gatsby/src/query/__tests__/query-compiler.js

```diff
@@ -96,3 +111,1181 @@ describe(`resolveThemes`, () => {
+      nodes.set(fileName, document)
+    })
+    const errors = []
+    const result = processQueries({
+      schema,
+      parsedQueries: nodes,
+      addError: e => {
+        errors.push(e)
+      },
+    })
+
+    expect(errors).toMatchInlineSnapshot(
+      `
+      Array [
+        Object {
+          "context": Object {
+            "closestFragment": "PostsJsonFragment",
+            "codeFrame": "   1 | query mockFileQuery {
+         2 |              allPostsJson {
+         3 |                nodes {
+      >  4 |                  ...PostJsonFragment
+           |                  ^^^^^^^^^^^^^^^^^^^
+         5 |                }
+         6 |             }
+         7 |           }
+         8 |` +
+        ` ` +
+        `
+         9 |           fragment PostsJsonFragment on PostsJson {
+        10 |             id
+        11 |           }",
+            "fragmentName": "PostJsonFragment",
+          },
+          "filePath": "mockFile",
+          "id": "85908",
+        },
+      ]
+    `
+    )
+    expect(result).toEqual(new Map())
+  })
+
+  it(`accepts identical fragment definitions`, async () => {
+    const nodes = new Map()
+    ;[
+      [
+        `mockFile`,
+        createGatsbyDoc(
+          `query mockFileQuery {
+           allPostsJson {
+             nodes {
+               ...PostsJsonFragment
+             }
+          }
+        }
+
+        fragment PostsJsonFragment on PostsJson {
+          id
+        }`
+        ),
+      ],
+      [
+        `mockComponent`,
+        createGatsbyDoc(
+          `fragment PostsJsonFragment on PostsJson {
+            id
+          }`
+        ),
+      ],
+    ].forEach(([fileName, document]) => {
+      nodes.set(fileName, document)
+    })
+    const errors = []
+    const result = processQueries({
+      schema,
+      parsedQueries: nodes,
+      addError: e => {
+        errors.push(e)
+      },
+    })
+    expect(errors).toEqual([])
+    expect(result).toMatchSnapshot()
+  })
+
+  it(`errors on duplicate fragment names`, async () => {
+    const nodes = new Map()
+    ;[
+      [
+        `mockFile`,
+        createGatsbyDoc(
+          `query mockFileQuery {
+           allPostsJson {
+             nodes {
+               ...PostsJsonFragment
+             }
+          }
+        }
+
+        fragment PostsJsonFragment on PostsJson {
+          id
+          node
+        }`
+        ),
+      ],
+      [
+        `mockComponent`,
+        createGatsbyDoc(
+          `fragment PostsJsonFragment on PostsJson {
+            id
+          }`
+        ),
+      ],
+    ].forEach(([fileName, document]) => {
+      nodes.set(fileName, document)
+    })
+    const errors = []
+    const result = processQueries({
+      schema,
+      parsedQueries: nodes,
+      addError: e => {
+        errors.push(e)
+      },
+    })
+    expect(errors).toMatchInlineSnapshot(
+      `
+      Array [
+        Object {
+          "context": Object {
+            "fragmentName": "",
```

Comment:
should this be here set the name?
```suggestion
            "fragmentName": "PostsJsonFragment",
```

Error `85919` expect an `fragmentName`:
`Found two different GraphQL fragments with identical name "${context.fragmentName}"`